docs/articles_en/learn-openvino/llm_inference_guide/genai-guide/genai-use-cases.rst

```diff
@@ -0,0 +1,305 @@
+                       ov::genai::height(896),
+                       ov::genai::num_inference_steps(20));
+                   imwrite("lora.bmp", image, true);
+
+                   std::cout << "Generating image without LoRA adapters applied, resulting image will be in baseline.bmp\n";
+                   image = pipe.generate(prompt,
+                       ov::genai::adapters(),  // Passing adapters as generation overrides set in the constructor; adapters() means no adapters.
+                       ov::genai::random_generator(std::make_shared<ov::genai::CppStdGenerator>(42)),
+                       ov::genai::width(512),
+                       ov::genai::height(896),
+                       ov::genai::num_inference_steps(20));
+                   imwrite("baseline.bmp", image, true);
+
+                   return EXIT_SUCCESS;
+               }
+
+
+      For more information, refer to the
+      `C++ sample <https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/cpp/text2image/README.md>`__
+
+Using GenAI in Speech Recognition
+#################################
+
+
+The application, shown in code samples below, performs inference on speech
+recognition Whisper Models. The samples include the ``WhisperPipeline`` class
+and use audio files in WAV format at a sampling rate of 16 kHz as input.
+
+.. tab-set::
+
+   .. tab-item:: Python
+      :sync: cpp
+
+      .. code-block:: python
+
+         import argparse
+         import openvino_genai
+         import librosa
+
+
+         def read_wav(filepath):
+             raw_speech, samplerate = librosa.load(filepath, sr=16000)
+             return raw_speech.tolist()
+
+
+         def main():
+             parser = argparse.ArgumentParser()
+             parser.add_argument("model_dir")
+             parser.add_argument("wav_file_path")
+             args = parser.parse_args()
+
+             raw_speech = read_wav(args.wav_file_path)
+
+             pipe = openvino_genai.WhisperPipeline(args.model_dir)
+
+             def streamer(word: str) -> bool:
+                 print(word, end="")
+                 return False
+
+             result = pipe.generate(
+                 raw_speech,
+                 max_new_tokens=100,
+                 # The 'task' and 'language' parameters are supported for multilingual models only.
+                 language="<|en|>",
+                 task="transcribe",
+                 return_timestamps=True,
+                 streamer=streamer,
+             )
+
+             print()
+
+             for chunk in result.chunks:
+                 print(f"timestamps: [{chunk.start_ts}, {chunk.end_ts}] text: {chunk.text}")
+
+
+      For more information, refer to the
+      `Python sample <https://github.com/openvinotoolkit/openvino.genai/blob/master/samples/python/whisper_speech_recognition/README.md>`__.
+
+   .. tab-item:: C++
+      :sync: cpp
+
+      .. code-block:: cpp
+
+         int main(int argc, char* argv[]) {
```

Comment:
Please use the built-in code from the code sample

Reply:
Can we reach a decision here? @slyalin I am taking into consideration your comment regarding distilling the code snippets.

Reply:
Currently, we see 2 possible solutions.

1. We can keep the whole built-in code, while commenting out some lines of code, thus leaving the part on which users can focus.
![image](https://github.com/user-attachments/assets/3af12a8c-3e30-4f11-a3d0-0a1eb4ac7608)


2. We keep the distilled code and provide a link to the source example. 

![image](https://github.com/user-attachments/assets/1e56b505-9173-49b4-b2d8-fa3cc7a13b86)

Please, share your thoughts.

Reply:
Or is it called `built-in code`?

Reply:
> Is there an option to embed the sample from a repo? Avoiding copy would be the best solution

As far as I know, it is impossible to directly embed files outside the scope of `openvino/docs` directory. We can make reference to such files.

Reply:
In that case having such commented out code is weird to me, so distilled code looks is better. @andrei-kochin will take a look once more.

Would there be some mechanism ensuring the code snippets are kept up to date?

Reply:
> In that case having such commented out code is weird to me, so distilled code looks is better. @andrei-kochin will take a look once more.
> 
> Would there be some mechanism ensuring the code snippets are kept up to date?

For now, I believe, these snippets will be listed under usual maintenance items to keep them up to date.

Reply:
@sgolebiewski-intel I agree to keep only inline comment related to the device. Others are not required and can be removed